src/native/util/os_linux.cpp

```diff
@@ -43,9 +43,14 @@ const std::vector<gid_t> ThreadScope::orig_groups = [] {
 void
 ThreadScope::change_user()
 {
+    if (_groups == NULL || _groups->empty()) {
+        MUST_SYS(syscall(SYS_setgroups, 0, NULL));
+    }
+    else if (_groups != NULL && *_groups != orig_groups) {
+        MUST_SYS(syscall(SYS_setgroups, _groups->size(), &_groups->at(0)));
```

Comment:
are we setting only the first item?

Reply:
no, cpp vectoe implementation is essentially wrapped c array. so sending a pointer to the first element is the same as sending pointer to a c array

Reply:
in c arrays are just contiguous memory. so it handles it by sending pointer to the beginning of the array, and it infers the offset by the type. maybe I can convert it to c array first, just to make it more readable